se_shared/include/se/path_planning/candidate_view.hpp

```diff
@@ -222,9 +228,9 @@ Eigen::Vector3i CandidateView<T>::getOffsetCandidate(const Eigen::Vector3i &cand
             << " offset by " << offset_v << " results in " << offset_cand_v.x() << " "
             << offset_cand_v.y() << " " << offset_cand_v.z() << " voxel state "
             << volume_._map_index->get(offset_cand_v).st;*/
-
+  int free_sphere = collision_check.isSphereCollisionFree(offset_cand_v);
   if (is_valid) {
-    if (collision_check.isSphereCollisionFree(offset_cand_v)) {
+    if (free_sphere==1) {
       return offset_cand_v;
```

Comment:
Calling `collision_check.isSphereCollisionFree` can be moved inside `if (is_valid)` to avoid some unnecessary calls.